File: Languages/Lua/math.lua
-- Hello, World!
print("Hello, World!")

print("\nProcessing math equations...")

function evaluate_expression(expr)
    -- Remove extra whitespace
    expr = expr:match("^%s*(.-)%s*$") or expr
    
    -- Use load to evaluate the expression safely
    local func, err = load("return " .. expr)
    if func then
        local success, result = pcall(func)
        if success then
            return result
        end
    end
    return "Error"
end

function solve_equation(line)
    local equation = line
    
    -- Remove "= ?" part
    local equals_pos = equation:find("=")
    if equals_pos then
        equation = equation:sub(1, equals_pos - 1)
    end
    
    -- Trim whitespace
    equation = equation:match("^%s*(.-)%s*$") or equation
    
    local result = evaluate_expression(equation)
    print(equation .. " = " .. tostring(result))
end

function process_file(filename)
    local file = io.open(filename, "r")
    if not file then
        print("Could not open file: " .. filename)
        return
    end
    
    for line in file:lines() do
        -- Skip empty lines and markdown headers
        if line ~= "" and not line:match("^#") then
            -- Handle markdown list items
            local start = line
            if line:match("^%- ") then
                start = line:sub(3)
            end
            
            if start:find("=") then
                solve_equation(start)
            end
        end
    end
    
    file:close()
end

-- Process all files
process_file("../test_data/math_equations.txt")
process_file("../test_data/math_equations.md")
process_file("../test_data/math_equations.json")
process_file("../test_data/math_equations.yaml")
process_file("../test_data/math_equations")

File: Languages/Rust/math.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

// Hello, World!
fn hello() {
    println!("Hello, World!");
}

fn evaluate_expression(expr: &str) -> Result<f64, Box<dyn std::error::Error>> {
    let parts: Vec<&str> expr.trim().split_whitespace().collect();
    
    if parts.len() < 3 {
        return Err("Invalid expression".into());
    }
    
    let a: f64 = parts[0].parse()?;
    let op = parts[1];
    let b: f64 = parts[2].parse()?;
    
    match op {
        "+" => Ok(a + b),
        "-" => Ok(a - b),
        "*" => Ok(a * b),
        "/" => {
            if b == 0.0 {
                Err("Division by zero".into())
            } else {
                Ok(a / b)
            }
        }
        _ => Err("Unknown operator".into()),
    }
}

fn solve_equation(line: &str) {
    let mut equation = line.to_string();
    
    // Remove "= ?" part
    if let Some(equals_pos) = equation.find('=') {
        equation.truncate(equals_pos);
    }
    
    // Trim whitespace
    equation = equation.trim().to_string();
    
    match evaluate_expression(&equation) {
        Ok(result) => println!("{} = {:.2}", equation, result),
        Err(_) => println!("{} = Error", equation),
    }
}

fn process_file(filename: &str) -> Result<(), Box<dyn std::error::Error>> {
    let path = Path::new(filename);
    let file = File::open(path)?;
    let reader = io::BufReader::new(file);
    
    for line in reader.lines() {
        let line = line?;
        
        // Skip empty lines and markdown headers
        if line.is_empty() || line.starts_with('#') {
            continue;
        }
        
        // Handle markdown list items
        let start = if line.starts_with("- ") {
            &line[2..]
        } else {
            &line
        };
        
        if start.contains('=') {
            solve_equation(start);
        }
    }
    
    Ok(())
}

fn main() {
    hello();
    
    println!("\nProcessing math equations...");
    
    let files = [
        "../test_data/math_equations.txt",
        "../test_data/math_equations.md",
        "../test_data/math_equations.json",
        "../test_data/math_equations.yaml",
        "../test_data/math_equations",
    ];
    
    for file in &files {
        if let Err(e) = process_file(file) {
            println!("Could not process file {}: {}", file, e);
        }
    }
}
